core-java/src/main/java/org/spine3/server/procman/ProcessManager.java

```diff
@@ -0,0 +1,288 @@
+        final EventContext.Builder builder = EventContext.newBuilder()
+                .setEventId(eventId)
+                .setVersion(currentVersion);
+
+        addEventContextAttributes(builder, commandId, event, currentState, currentVersion);
+
+        return builder.build();
+    }
+
+    /**
+     * Adds custom attributes to an event context builder during the creation of the event context.
+     *
+     * <p>Does nothing by default. Override this method if you want to add custom attributes to the created context.
+     *
+     * @param builder        a builder for the event context
+     * @param commandId      the id of the command, which cased the event
+     * @param event          the event message
+     * @param currentState   the current state of the aggregate after the event was applied
+     * @param currentVersion the version of the process manager after the event was applied
+     * @see #createEventContext(CommandId, Message, Message, Timestamp, int)
+     */
+    @SuppressWarnings({"NoopMethodInAbstractClass", "UnusedParameters"}) // Have no-op method to avoid forced overriding.
+    protected void addEventContextAttributes(EventContext.Builder builder,
+                                             CommandId commandId, Message event, M currentState, int currentVersion) {
+        // Do nothing.
+    }
+
+    /**
+     * Returns the set of the command types handled by the process manager.
+     *
+     * @param pmClass the process manager class to inspect
+     * @return immutable set of command classes or an empty set if no commands are handled
+     */
+    public static Set<Class<? extends Message>> getHandledCommandClasses(Class<? extends ProcessManager> pmClass) {
+        return Classes.getHandledMessageClasses(pmClass, CommandHandlerMethod.isCommandHandlerPredicate);
+    }
+
+    /**
+     * Returns the set of event classes handled by the process manager.
+     *
+     * @param pmClass the process manager class to inspect
+     * @return immutable set of event classes or an empty set if no events are handled
+     */
+    public static ImmutableSet<Class<? extends Message>> getHandledEventClasses(Class<? extends ProcessManager> pmClass) {
+        return Classes.getHandledMessageClasses(pmClass, EventHandlerMethod.isEventHandlerPredicate);
+    }
+
+    private IllegalStateException missingCommandHandler(Class<? extends Message> commandClass) {
+        return new IllegalStateException(String.format("Missing handler for command class %s in process manager class %s.",
+                        commandClass.getName(), getClass().getName()));
+    }
+
+    private IllegalStateException missingEventHandler(Class<? extends Message> eventClass) {
+        return new IllegalStateException(String.format("Missing event handler for event class %s in the process manager class %s",
+                eventClass, this.getClass()));
+    }
+
+    /**
+     * The registry of method maps for all process manager classes.
+     * <p>
```

Comment:
Have <p> before the start of the paragraph and an empty line in between first line and the next one. Similarly to what we do in the rest of class docs.